File: utils/session.py
# Foydalanuvchi joriy tilini in-memory(xotirada) saqlash uchun lug'at
# Katta masshtab uchun buni FSM Storage, Redis yoki bazaga(Postgres, SQLite) ko'chirgan ma'qul.

user_langs = {}

def get_user_lang(user_id: int) -> str:
    """
    Foydalanuvchi tanlagan tilni qaytaradi.
    Agar til u qadar tanlanmagan bo'lsa, avtomatik ravishda 'uz' qaytadi.
    """
    return user_langs.get(user_id, 'uz')

def set_user_lang(user_id: int, lang: str):
    """
    Foydalanuvchi tilini sessiyaga saqlaydi.
    """
    user_langs[user_id] = lang

File: cf_bot/js_bot/utils/session.js
// Foydalanuvchi joriy tilini in-memory(xotirada) saqlash uchun lug'at
const user_langs = new Map();

/**
 * Foydalanuvchi tanlagan tilni qaytaradi.
 * Agar til u qadar tanlanmagan bo'lsa, avtomatik ravishda 'uz' qaytadi.
 * @param {number} userId 
 * @returns {string} 
 */
function getUserLang(userId) {
    if (user_langs.has(userId)) {
        return user_langs.get(userId);
    }
    return 'uz';
}

/**
 * Foydalanuvchi tilini sessiyaga saqlaydi.
 * @param {number} userId 
 * @param {string} lang 
 */
function setUserLang(userId, lang) {
    user_langs.set(userId, lang);
}

module.exports = {
    getUserLang,
    setUserLang
};
